src/CattyTests/SceneTests.swift

```diff
@@ -123,4 +123,40 @@ final class SceneTests: XCTestCase {
         XCTAssertEqual(0, userVariable.textLabel?.text?.count)
         scene.stopProject()
     }
+
+    func testUpdate() {
+
+        class DummyCBSpriteNode: CBSpriteNode {
+
+            var methodCallCount = 0
+
+            override func update(_ currentTime: TimeInterval) {
+                self.methodCallCount += 1
+            }
+
+        }
+
+        let scene = SceneBuilder(project: ProjectMock()).build()
+        let spriteNode = DummyCBSpriteNode(spriteObject: SpriteObject())
+        spriteNode.name = "testObject1"
+        scene.scheduler.registerSpriteNode(spriteNode)
+
+        var expectedCallCount = 0
+        XCTAssertEqual(spriteNode.methodCallCount, expectedCallCount)
+
+        for i in 1...6 {
```

Comment:
Very good test! However a little bit hard to read because of the calculations. My suggestion would be to execute the `update` function manually four times without any loops and ifs. Also please make use of a second `SpriteNode` to verify that all nodes are "updated".